### ISSUE TITLE
cli / upgrade_database.php error

### ISSUE THREAD

**Issue by dabeani** [Cacti/cacti#1421]
hi,

get some error message while execute cli/upgrade_database.php...

>php upgrade_database.php 
PHP Notice:  Undefined variable: new_modes in /usr/share/cacti/lib/database.php on line 92
PHP Warning:  implode(): Invalid arguments passed in /usr/share/cacti/lib/database.php on line 92
Your Cacti is already up to date.

br
Bernhard

**Comment by dabeani** [comment]
same execution error will popup while executing cli/repait_templates.php!

php repair_templates.php 
PHP Notice:  Undefined variable: new_modes in /usr/share/cacti/lib/database.php on line 92
PHP Warning:  implode(): Invalid arguments passed in /usr/share/cacti/lib/database.php on line 92
NOTE: Performing Check of Templates
NOTE: Performing Check of Data Templates
NOTE: No Damaged Data Templates Found
NOTE: Performing Check of Graph Templates
NOTE: No Damaged Graph Templates Found

**Comment by anarkia1976** [comment]
No problem here, do you have the right version of php?

**Comment by netniV** [comment]
The error can safely be ignored.  What it means is that there isn't an SQL MODE that cacti doesn't like... so weirdly it's an error about there NOT being a problem ;-)

```php
// Get rid of bad modes
$modes = explode(',', db_fetch_cell('SELECT @@sql_mode'))
foreach($modes as $mode) {
        if (array_search($mode, $bad_modes) === false) {
                $new_modes[] = $mode;
        }
}
$sql_mode = implode(',', $new_modes);
db_execute('SET SESSION sql_mode = "' . $sql_mode . '"');
```

The simply fix is to add new_modes = array() and then check the sizeof() before issuing the corrective db_execute statement.

**Comment by netniV** [comment]
Wait no, what that actually means is, all your modes are disliked by cacti.  So, just adding  the array() primer will fix the issue.

**Comment by netniV** [comment]
This has now been merged into the develop branch via PR #1425
